### DISCUSSION TITLE
How to detect a remote scanner?

### DISCUSSION THREAD

**Discussion by stefanofornari** [cyanfish/naps2#251] [Support]
Hello,
my scanner is attached to a linux server whose share it through SANE. I can use the scanner (HP DeskJet 1510) from my ubuntu laptop, but NAPS2 does not detect it.

Is this configuration supported? command line would work as well...

**Comment by cyanfish** [comment]
If it's usable through SANE at all (e.g. via scanimage on the command line) it should be usable in NAPS2. One thing to check is that you're waiting long enough for NAPS2 to finish searching for devices (sometimes 10s+), there's a loading icon but it doesn't animate properly in some desktop environments.

If the scanner can't be found even in scanimage then you'll need to debug the issue on the SANE end. If it shows in scanimage but not NAPS2 still, if you're using the flatpak you can try the deb/rpm instead.

An alternative to sharing via SANE is sharing via the recently-added [scanner sharing](https://github.com/cyanfish/naps2/blob/master/CHANGELOG.md) feature in NAPS2.

**Reply by stefanofornari** [reply]
Thanks for replying first of all. I waited long enough but it could not find it. But I think I have also an idea why: there is not a way to tell NAPS2 which server attach to (it could scan the local network, but it may be not really efficient...). On my laptop, I had to set it in /etc/sane.d/net.conf, e.g: 

```
cat /etc/sane.d/net.conf 
# This is the net backend config file.

## net backend options
# Timeout for the initial connection to saned. This will prevent the backend
# from blocking for several minutes trying to connect to an unresponsive
# saned host (network outage, host down, ...). Value in seconds.
connect_timeout = 60

## saned hosts
# Each line names a host to attach to.
# If you list "localhost" then your backends can be accessed either
# directly or through the net backend.  Going through the net backend
# may be necessary to access devices that need special privileges.
# localhost
192.168.0.10
```

I would expect something similar.
